public/app/containers/Explore/PromQueryField.tsx

```diff
@@ -409,6 +437,38 @@ class PromQueryField extends React.Component<PromQueryFieldProps, PromQueryField
     });
   }
 
+  // Temporarily here while reusing this field for logging
+  async fetchLogLabels() {
+    const url = '/api/prom/label';
+    try {
+      const res = await this.request(url);
+      const body = await (res.data || res.json());
+      const labelKeys = body.data.slice().sort();
+      const labelKeysBySelector = {
+        ...this.state.labelKeys,
+        [EMPTY_SELECTOR]: labelKeys,
+      };
+      const labelValuesByKey = {};
+      const logLabelOptions = [];
+      for (const key of labelKeys) {
+        const valuesUrl = `/api/prom/label/${key}/values`;
+        const res = await this.request(valuesUrl);
```

Comment:
this looks very expensive to have this initial request and the a for loop that spawns possibly hundreds of requests ?

Reply:
Yes, I know. I've already requested an API endpoint to be added to logging that returns the whole label tree at once. In the meantime, I think I can rewrite this to lazy-load the values when a key is selected in the picker.